src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,116 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private int size;
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_SIZE];
+    }
 
     @Override
     public void add(T value) {
-
+        if (array.length == size) {
+            array = grow();
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+        } else if (index < 0 || index > size) {
+            throw new ArrayIndexOutOfBoundsException("Add element failed. Index out of bounds");
+        } else {
+            if (size == array.length) {
+                array = grow();
+            }
+            for (int i = size - 1; i >= index; i--) {
+                moveRight(i);
+            }
```

Comment:
use System.arraycopy() method. Same in remove

Reply:
Let's make checks in this order:
check index range(throw exception;
if index = size use add();
add by index  with shift of all elements;
Dont use else block after throw keyword. It's redundant